underfs/local/src/main/java/alluxio/underfs/local/LocalUnderFileSystem.java

```diff
@@ -229,19 +238,23 @@ public long getSpace(String path, SpaceType type) throws IOException {
   public UfsStatus getStatus(String path) throws IOException {
     String tpath = stripPath(path);
     File file = new File(tpath);
-    PosixFileAttributes attr =
-        Files.readAttributes(Paths.get(file.getPath()), PosixFileAttributes.class);
-    if (file.isFile()) {
-      // Return file status.
-      String contentHash =
-          UnderFileSystemUtils.approximateContentHash(file.length(), file.lastModified());
-      return new UfsFileStatus(path, contentHash, file.length(), file.lastModified(),
-          attr.owner().getName(), attr.group().getName(),
-          FileUtils.translatePosixPermissionToMode(attr.permissions()));
+    try {
+      PosixFileAttributes attr =
+          Files.readAttributes(Paths.get(file.getPath()), PosixFileAttributes.class);
+      if (file.isFile()) {
+        // Return file status.
+        String contentHash =
+            UnderFileSystemUtils.approximateContentHash(file.length(), file.lastModified());
+        return new UfsFileStatus(path, contentHash, file.length(), file.lastModified(),
+            attr.owner().getName(), attr.group().getName(),
+            FileUtils.translatePosixPermissionToMode(attr.permissions()));
+      }
+      // Return directory status.
+      return new UfsDirectoryStatus(path, attr.owner().getName(), attr.group().getName(),
+          FileUtils.translatePosixPermissionToMode(attr.permissions()), file.lastModified());
+    } catch (FileSystemException e) {
```

Comment:
Is it safe to convert `FileSystemException` to `FileNotFound`, should we catch `NoSuchFile`/`NoSuchDirectory` instead?

Reply:
that's what i had earlier but then changed to `FileSystemException`. if `/dir` is a file, and we `getStatus(/dir/file` local ufs throws  `FileSystemException` instead of `NoSuchFileException` at readAttributes